core/src/main/java/com/homeaway/streamplatform/streamregistry/StreamRegistryApplication.java

```diff
@@ -124,8 +132,23 @@ public void run(final StreamRegistryConfiguration configuration, final Environme
         kstreamsProperties.put(ROCKSDB_CONFIG_SETTER_CLASS_CONFIG, CustomRocksDBConfig.class);
         TopicsConfig topicsConfig = configuration.getTopicsConfig();
 
-        ManagedKafkaProducer managedProducer = new ManagedKafkaProducer(producerProperties, topicsConfig);
-        ManagedKStreams managedKStreams = new ManagedKStreams(kstreamsProperties, topicsConfig);
+        StreamRegistryProducer<AvroStreamKey, AvroStream> streamProducer = new StreamRegistryProducer(producerProperties, topicsConfig.getProducerTopic());
+        StreamRegistryProducer<AvroStreamKey, Sources> sourceProducer = new StreamRegistryProducer(producerProperties, topicsConfig.getStreamSourceTopic());
+
+        Properties streamProperties = new Properties();
+        streamProperties.putAll(kstreamsProperties);
+        streamProperties.put(StreamsConfig.STATE_DIR_CONFIG, "/tmp/streams/kafka-streams");
+
+        GlobalKStreams<AvroStream> streamProcessor = new GlobalKStreams(streamProperties,
+                topicsConfig.getProducerTopic(), topicsConfig.getProducerStateStore(), null);
+
+
+        Properties sourceProperties = new Properties();
+        sourceProperties.putAll(kstreamsProperties);
+        sourceProperties.put(StreamsConfig.STATE_DIR_CONFIG, "/tmp/sources/kafka-streams");
```

Comment:
Should the state directories be defined in the config file?

Reply:
Yeah. Probably a good idea. Will do.